HashTable.java

```diff
@@ -0,0 +1,107 @@
+class HashTable{
+
+    //I am using String array as a practice to see what happens when Strings are stored in HashMap/HashTable
+    String[] hashArr;
+    int size;
+    int arrSize;
+
+    public HashTable(int numOfSlots){
+        //This i read in some documentation that it is always better to have prime number as slots.
+        //The main reason can be due to the use of mod operator to calculate the index, prime number helps.
+
+        //So I am going to check if the number of slots are prime. if not then i will check the next available
+        //prime number and assign it to the the slot.
+        if (isPrime(numOfSlots)) {
+            hashArr = new String[numOfSlots];
+            arrSize = numOfSlots;
+        }
+        else{
+            int nextPrimeNumber = getNextPrimeNumber(numOfSlots);
+            hashArr = new String[nextPrimeNumber];
+            arrSize = nextPrimeNumber;
+        }
+    }
+
+    private boolean isPrime(int numOfSlots){
+
+        for (int i = 2; i*i <= numOfSlots; i++){
+            if (numOfSlots % i == 0){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private int getNextPrimeNumber(int numOfSlots){
+        int i = numOfSlots;
+        while (true){
+            if (isPrime(i)){
+                return i;
+            }
+            i++;
+        }
+    }
+
+    /*
+    Now that we have proper size of the array, we need to have hashfunctions for double hashing.
+    We will have two hash functions. First one will find a slot. If the element is already present
+    then we will use one more hashfunction which will skip the clusters/clumps and try to insert the
+    element in new position
+     */
+
+    private int hashFunction1(String word){
+        int index = word.hashCode();
+        index = index % arrSize;
+
+        //sometimes Java can return negative hashcode, hence  we will try to avoid this
+        if (index < 0){
+            index += arrSize;
```

Comment:
What is the guarantee that by incrementing index by arrSize once it will become a positive integer?